### ISSUE TITLE
Wolverine with RabbitMQ publishes only 1 out of 2 messages

### ISSUE THREAD

**Issue by da-kil** [JasperFx/wolverine#1181]
**Describe the bug**
In my sample app I create two object (WolverineTest.ApiService.Program.cs line 40-53). I insert the object into the session, call outbox.PublishAsync and session.SaveChangesAsync. Both objects are inserted to the DB table postgres.mt_doc_weatherforecast.
I see one entry in issue_events.wolverine_outgoing_envelopes and one entry in issue_events.wolverine_incoming_envelopes with status 'handled'. The test handler (WolverineTest.ApiService.Program.cs line 125-130) is triggered only once. I would expect two calls. If I restart the solution the second event gets handled.

**To Reproduce**
1. Set 'WolverineTest.AppHost' as startup project
2. Start debbuging
3. Wait until the aspire dashboard is loaded completely
4. Open webfrontend (usually last entry)
5. Click on 'Trigger', now you see the behaviour as described above

[WolverineTest.zip](https://github.com/user-attachments/files/18188663/WolverineTest.zip)

**Comment by jeremydmiller** [comment]
@da-kil You're only publishing one message in that code.

**Comment by da-kil** [comment]
@jeremydmiller that's not true, I'm looping twice, I also have to two entries in marten db.
`for (int i = 0; i < 2; i++) {}`
